lib/gears/matcher.lua

```diff
@@ -0,0 +1,292 @@
+--- A module to build a set of properties based on a graph of rules.
+--
+-- Rules come from a graph of sources. Each source depend on other and must be
+-- applied before other sources. For this module to work, the graph needs to be
+-- solvable.
+--
```

Comment:
Do rules actually depend on others?  What about the root?

Reply:
> Do rules actually depend on others?

This is unchanged from `awful.rules`. If different *properties* have a dependency on one another, then they must be in different rule sources. Otherwise the array order is used.